### PR TITLE
feat: CDI injection support for Routes

### PR DESCRIPTION
**Pull request by caalador**
Add support for CDI injection.

### CODE REVIEW

File: deployment/src/main/java/com/vaadin/quarkus/deployment/VaadinQuarkusProcessor.java [hunk 1]
```diff
@@ -15,29 +15,103 @@
...
+
+        // If no VaadinServlet instances found register QuarkusVaadinServlet
+        if (vaadinServlets.isEmpty()) {
+            servletProducer.produce(ServletBuildItem
+                    .builder(QuarkusVaadinServlet.class.getName(),
+                            QuarkusVaadinServlet.class.getName())
+                    .addMapping("/*").setAsyncSupported(true).build());
+        } else {
+            // TODO: check that we don't register 2 of the same mapping
+            for (ClassInfo info : vaadinServlets) {
```
Comment (denis-anisimov) [inline deployment/src/main/java/com/vaadin/quarkus/deployment/VaadinQuarkusProcessor.java:88]: quite long block inside `else` .
I would refactor it into  separate method(s) to simplify reading

File: runtime/src/main/java/com/vaadin/quarkus/QuarkusInstantiator.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+public class QuarkusInstantiator implements Instantiator {
+
+    private static final String CANNOT_USE_CDI_BEANS_FOR_I18N = "Cannot use CDI beans for I18N, falling back to the default behavior.";
+    private static final String FALLING_BACK_TO_DEFAULT_INSTANTIATION = "Falling back to default instantiation.";
+
+    private AtomicBoolean i18NLoggingEnabled = new AtomicBoolean(true);
+    private DefaultInstantiator delegate;
+
+    @Inject
+    BeanManager beanManager;
```
Comment (denis-anisimov) [inline runtime/src/main/java/com/vaadin/quarkus/QuarkusInstantiator.java:48]: `private` ?

File: runtime/src/main/java/com/vaadin/quarkus/QuarkusVaadinServlet.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.vaadin.flow.function.DeploymentConfiguration;
+import com.vaadin.flow.server.ServiceException;
+import com.vaadin.flow.server.VaadinServlet;
+import com.vaadin.flow.server.VaadinServletService;
+
+public class QuarkusVaadinServlet extends VaadinServlet {
+
+    @Inject
+    BeanManager beanManager;
```
Comment (denis-anisimov) [inline runtime/src/main/java/com/vaadin/quarkus/QuarkusVaadinServlet.java:29]: `private`

**Review by denis-anisimov — CHANGES_REQUESTED** [review]

File: runtime/src/main/java/com/vaadin/quarkus/QuarkusInstantiator.java (hunk 2 above)
Comment (caalador) [reply runtime/src/main/java/com/vaadin/quarkus/QuarkusInstantiator.java:48]: It's as per Quarkus usage recommendation. Havin it as private will on runtime print:
```
[INFO] [io.quarkus.arc.processor.BeanProcessor] Found unrecommended usage of private members (use package-private instead) in application beans:
	- @Inject field com.vaadin.quarkus.QuarkusVaadinServlet#beanManager,
```

File: runtime/src/main/java/com/vaadin/quarkus/QuarkusVaadinServlet.java (hunk 3 above)
Comment (caalador) [reply runtime/src/main/java/com/vaadin/quarkus/QuarkusVaadinServlet.java:29]: It's as per Quarkus usage recommendation.

File: deployment/src/main/java/com/vaadin/quarkus/deployment/VaadinQuarkusProcessor.java (hunk 1 above)
Comment (caalador) [reply deployment/src/main/java/com/vaadin/quarkus/deployment/VaadinQuarkusProcessor.java]: done

File: runtime/src/main/java/com/vaadin/quarkus/QuarkusInstantiator.java (hunk 2 above)
Comment (denis-anisimov) [reply runtime/src/main/java/com/vaadin/quarkus/QuarkusInstantiator.java:48]: :(

File: deployment/src/main/java/com/vaadin/quarkus/deployment/VaadinQuarkusProcessor.java [hunk 4]
```diff
@@ -15,29 +15,108 @@
...
+            servletProducer.produce(ServletBuildItem
+                    .builder(QuarkusVaadinServlet.class.getName(),
+                            QuarkusVaadinServlet.class.getName())
+                    .addMapping("/*").setAsyncSupported(true).build());
+        } else {
+            registerUserServlets(servletProducer, vaadinServlets);
+        }
+    }
+
+    private void registerUserServlets(BuildProducer<ServletBuildItem> servletProducer,
```
Comment (denis-anisimov) [inline deployment/src/main/java/com/vaadin/quarkus/deployment/VaadinQuarkusProcessor.java:91]: the method still should be refactored: `for` doesn't have to be in the method, the inner block may be a method.

The method impl contains `for` inside `for` and `if` inside `for` and the `for` block is quite hard to read.

**Comment by caalador** [comment]
Split to methods and rebased PR on master to get correct test changes.

**Review by denis-anisimov — APPROVED** [review]

**Pull request merged**